kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/util/ColumnValueConvertUtils.java

```diff
@@ -108,9 +103,7 @@ public static Message convertToProtobufMessage(final Object object) {
         }
         if (object instanceof Time) {
             Time time = (Time) object;
-            long millis = (int) (time.getTime() % MILLISECONDS_PER_SECOND);
-            int nanosOfSecond = (int) (millis * NANOSECONDS_PER_MILLISECOND);
-            LocalTime localTime = LocalTime.of(time.getHours(), time.getMinutes(), time.getSeconds(), nanosOfSecond);
+            LocalTime localTime = LocalTime.of(time.getHours(), time.getMinutes(), time.getSeconds(), new Timestamp(time.getTime()).getNanos());
```

Comment:
`new Timestamp()` will fix nano is negative value
<img width="726" alt="image" src="https://github.com/apache/shardingsphere/assets/101622833/bd559fe5-3101-4fa3-bdc4-3deeef49e317">